upgrade/runner.py

```diff
@@ -236,11 +234,16 @@ def product_upgrade(product):
                                 foreman_debug, 'capsule_{}'.format(cap_host),
                                 host=cap_host)
                             raise
+                    # Execute tasks as post upgrade tier1 tests are
+                    # dependent
+                    post_upgrade_test_tasks(sat_host, cap_hosts)
```

Comment:
You need not have it here, which will unnecessary run tasks e.g katello-service restart on satellite which you don't want to run here. You want to run them after all upgrade for tier tests.

Also, you are already checking if cap_hosts are given and then only you are running capsule sync related task.

So remove this from here and keep all the post-upgrade tasks after all upgrade

Reply:
ahh, right, My bad.